neurovault/apps/statmaps/views.py

```diff
@@ -650,6 +658,60 @@ def atlas_query_voxel(request):
         return JSONResponse('error: one or more coordinates are out of range', status=400)
     return JSONResponse(data)
 
+# Compare Two Images
+def compare_images(request,pk1,pk2):
+    image1 = get_image(pk1,None,request)
+    image2 = get_image(pk2,None,request)
+    
+    # Name the data file based on the new volumes
+    path, name1, ext = split_filename(image1.file.url)
+    path, name2, ext = split_filename(image2.file.url)
+
+    # For now, manually choose an atlas - this can be user choice
+    atlas_file = os.path.abspath(os.path.join(neurovault.settings.STATIC_ROOT, 'atlas/MNI-maxprob-thr25-2mm.nii'))
+    atlas_xml = os.path.abspath(os.path.join(neurovault.settings.STATIC_ROOT, 'atlas/MNI.xml'))
+    atlas = pybrainatlas.atlas(atlas_xml,atlas_file) # Default slices are "coronal","axial","sagittal"
+
+    # Create custom image names for the visualization
+    custom = {"image 1":"%s:%s" %(image1.name,image1.map_type),"image 2": "%s:%s" %(image2.name,image2.map_type)}
+
+    html_snippet,data_table = compare.scatterplot_compare(image1=image1.file.path,image2=image2.file.path,software="FREESURFER",voxdim=[8,8,8],atlas=atlas,custom=custom,corr="pearson")
+
+    html = [h.strip("\n") for h in html_snippet]
+    context = {'html': html}
+    return render(request, 'statmaps/compare_images.html', context)
+
+# Return search interface for one image vs rest
+def find_similar(request,pk):
+    image1 = get_image(pk,None,request)
+    # This df should ONLY contain public images! see tasks.py for generation
+    corr_df = pandas.read_pickle(os.path.abspath(os.path.join(neurovault.settings.PRIVATE_MEDIA_ROOT,'matrices/pearson_corr.pkl')))
+    public_images = Image.objects.filter(collection__private=False,id__in=corr_df.columns)
+
+    # Get all image png paths
+    image_paths = [image.file.url for image in public_images]
+    png_img_names = ["glass_brain_%s.png" % image.pk for image in public_images]  
+    png_img_paths = [ os.path.join(os.path.split(image_paths[i])[0],png_img_names[i]) for i in range(0,len(image_paths))]
+    
+    # Get image tags, for now we will just do map type
+    tags = [[str(image.map_type)] for image in public_images]
+    
+    # Tags and png paths should be put in same data frame, so image ids are associated with both
+    corr_df["png"] = png_img_paths
+    corr_df["tags"] = tags
+    compare_url = "/compare" # format will be prefix/[query_id]/[other_id]
+    image_url = "/images" # format will be prefix/[other_id]
+    
+    # Here is the query image
+    query = os.path.join(os.path.split(image1.file.url)[0],"glass_brain_%s.png" %(image1.pk))
+    
+    # Do similarity search and return html to put in page
+    html_snippet = compare.similarity_search(corr_df=corr_df,button_url=compare_url,image_url=image_url,query=query)
```

Comment:
where in code can I control how many similar images are displayed? Are you including high negative correlations as well?